### PR TITLE
fix(replays): Replay list error UI

### LINKED ISSUE #38367: Bug: Invalid search inputs should show no replay results instead of an error view

### PR DESCRIPTION
**Pull request by danecando**
When running a search query that isn't supported the error UI replaced the entire replay list view. This update keeps the view/list UI and renders the error in the table.

Closes #38367 

### Changes
- Moves replay list fetch error UI inside of the table

### Notes
The regular issues/events API just returns no results when given invalid search args and the replay API returns a 400. So a difference in behavior between the two APIs. 

Before:

Screen Shot 2022-09-06 at 8 23 46 AM

After:

Screen Shot 2022-09-06 at 8 23 38 AM

Updated after with Alert:
Screen Shot 2022-09-06 at 12 57 06 PM

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: static/app/views/replays/replayTable.tsx [hunk 1]
```diff
@@ -72,46 +74,78 @@ function SortableHeader({
...
+            <div>
+              <p>{t('This could be due to a handful of reasons:')}</p>
+              <ol className="detailed-error-list">
+                {reasons.map((reason, i) => (
+                  <li key={i}>{reason}</li>
+                ))}
+              </ol>
+            </div>
+          }
+        />
```
Comment (billyvg) [inline static/app/views/replays/replayTable.tsx:138]: Let's use `<Alert>` - https://storybook.sentry.dev/?path=/story/components-alerts-alert--with-icons

Example on issue stream:
image


(although there shouldn't be a margin-bottom there)

**Review by billyvg — APPROVED** [review]

**Review by ryan953 — CHANGES_REQUESTED** [review]
nvm, can put all this in a new ticket and iterate!

**Review by ryan953 — APPROVED** [review]

File: static/app/views/replays/replayTable.tsx [hunk 2]
```diff
@@ -72,46 +74,63 @@ function SortableHeader({
...
+  if (fetchError && !isFetching) {
+    return (
+      <StyledPanelTable
+        headers={tableHeaders}
+        showProjectColumn={showProjectColumn}
+        isLoading={false}
+      >
+        <StyledAlert type="error" showIcon>
+          Sorry, the list of replays could not be loaded. This could be due to invalid
+          search parameters or an internal systems error.
```
Comment (billyvg) [inline static/app/views/replays/replayTable.tsx:122]: Oops this should be wrapped in a `t()`

**Comment by ryan953** [comment]
conflicts

**Pull request merged**